src/v2i-hub/RSUHealthMonitorPlugin/src/RSUHealthMonitorWorker.cpp

```diff
@@ -0,0 +1,175 @@
+
+namespace RSUHealthMonitor
+{
+
+    RSUHealthMonitorWorker::RSUHealthMonitorWorker()
+    {
+        _RSUSTATUSConfigMapPtr = make_shared<map<RSUMibVersion, RSUStatusConfigTable>>();
+        // Currently only support RSU MIB version 4.1. Other future supported versions will be inserted here.
+        RSUStatusConfigTable rsuRstatusTable = constructRsuStatusConfigTable(RSUMibVersion::RSUMIB_V_4_1);
+        _RSUSTATUSConfigMapPtr->insert({RSUMibVersion::RSUMIB_V_4_1, rsuRstatusTable});
+    }
+
+    RSUStatusConfigTable RSUHealthMonitorWorker::constructRsuStatusConfigTable(const RSUMibVersion &mibVersion) const
+    {
+        RSUStatusConfigTable rsuStatusTbl;
+        // Populate custome defined RSU Status table with RSU MIB version 4.1.
+        if (mibVersion == RSUMibVersion::RSUMIB_V_4_1)
+        {
+            RSUFieldOIDStruct rsuID = {"rsuID", RSU_ID_OID, true};
+            rsuStatusTbl.push_back(rsuID);
+
+            RSUFieldOIDStruct rsuMibVersion = {"rsuMibVersion", RSU_MIB_VERSION, true};
+            rsuStatusTbl.push_back(rsuMibVersion);
+
+            RSUFieldOIDStruct rsuFirmwareVersion = {"rsuFirmwareVersion", RSU_FIRMWARE_VERSION, true};
+            rsuStatusTbl.push_back(rsuFirmwareVersion);
+
+            RSUFieldOIDStruct rsuManufacturer = {"rsuManufacturer", RSU_MANUFACTURER, true};
+            rsuStatusTbl.push_back(rsuManufacturer);
+
+            RSUFieldOIDStruct rsuGpsOutputString = {"rsuGpsOutputString", RSU_GPS_OUTPUT_STRING, true};
+            rsuStatusTbl.push_back(rsuGpsOutputString);
+
+            RSUFieldOIDStruct rsuIFMIndex = {"rsuIFMIndex", RSU_IFM_INDEX, false};
+            rsuStatusTbl.push_back(rsuIFMIndex);
+
+            RSUFieldOIDStruct rsuIFMPsid = {"rsuIFMPsid", RSU_IFM_PSID, false};
+            rsuStatusTbl.push_back(rsuIFMPsid);
+
+            RSUFieldOIDStruct rsuIFMDsrcMsgId = {"rsuIFMDsrcMsgId", RSU_IFM_DSRC_MSG_ID, false};
+            rsuStatusTbl.push_back(rsuIFMDsrcMsgId);
+
+            RSUFieldOIDStruct rsuIFMTxMode = {"rsuIFMTxMode", RSU_IFM_INDEX, false};
+            rsuStatusTbl.push_back(rsuIFMTxMode);
+
+            RSUFieldOIDStruct rsuIFMTxChannel = {"rsuIFMTxChannel", RSU_IFM_TX_CHANNEL, false};
+            rsuStatusTbl.push_back(rsuIFMTxChannel);
+
+            RSUFieldOIDStruct rsuIFMEnable = {"rsuIFMEnable", RSU_IFM_ENABLE, false};
+            rsuStatusTbl.push_back(rsuIFMEnable);
+
+            RSUFieldOIDStruct rsuIFMStatus = {"rsuIFMStatus", RSU_IFM_STATUS, false};
+            rsuStatusTbl.push_back(rsuIFMStatus);
+
+            RSUFieldOIDStruct rsuMode = {"rsuMode", RSU_MODE, true};
+            rsuStatusTbl.push_back(rsuMode);
+
+            RSUFieldOIDStruct rsuChanStatus = {"rsuChanStatus", RSU_CHAN_STATUS, true};
+            rsuStatusTbl.push_back(rsuChanStatus);
+        }
+        return rsuStatusTbl;
+    }
+
+    bool RSUHealthMonitorWorker::isAllRequiredFieldsPresent(const RSUMibVersion &mibVersion, const vector<string> &fields) const
```

Comment:
If this is a validation method I suggest for clarity sake you pass in the config directly instead of passing in the mib version and calling Get config. I think it is cleaner for the validate method to take parameters ( message to validate, list of required fields)

Reply:
Addressed https://github.com/usdot-fhwa-OPS/V2X-Hub/pull/564/commits/a49366ea8030afa8c668ec5879cf8e2d5cf69065